2020/day5/solution.py

```diff
@@ -0,0 +1,41 @@
+import itertools
+
+class Ticket:
+    _row = None
+    _column = None
+
+    def __init__(self, code):
+        self._code = code
+    
+    @property
+    def seat_id(self):
+        return self.row * 8  + self.column
+    
+    @property
```

Comment:
Properties are great! In this case `seat_id is a great use of a property.
However, `row` and `column` don't need to be properties:
- `_row` and `_column` are never mutated after they are initially set
- computing `row` and `column` is not expensive
So, it will make more sense to compute & set `row` and `column` as regular variables directly in the constructor, no need for the underscore versions or properties. (You also don't need to store `_code` anymore!)

This also lets you do e.g.`row, column = code[:7], code[7:]` (and then set `self.row` and `self.column` to the parsed values) which makes it more clear you're splitting the input.

Reply:
This also smells like a good usage for [`dataclasses`](https://docs.python.org/3/library/dataclasses.html) (which is a newer/better version of the namedtuple trick) - it's great for POD classes.

Reply:
Actually, put the parsing code in a separate classmethod:
```
@classmethod
def from code(cls, code):
    ...
    return cls(row, column)
```
This will let you retain the auto-generated dataclass constructor which takes a row & column which is useful down below (see my other comment).

Reply:
ooh, i like the idea of a  class method that works as a constructor. I did want to have an actual row, column constructor and  then discovered python doesn't allow multiple  __init__ methods